k2/python/k2/fsa.py

```diff
@@ -791,6 +821,55 @@ def requires_grad_(self, requires_grad: bool) -> 'Fsa':
         self.scores.requires_grad_(requires_grad)
         return self
 
+    def rename_tensor_attribute_(self, src_name: str, dest_name: str) -> 'Fsa':
+        '''
+        Rename a tensor attribute, and also rename non-tensor
+        attributes that are associated with it, i.e. that have it
+        as a prefix.
+        Args:
+           src_name: The original name, must exist as
+                a tensor attribute, e.g. 'aux_labels'.
+           dest_name: The new name, that we are renaming it to.
+                If it already existed as a tensor attribute, it will
+                be rewritten; and any previously existing non-tensor
+                attributes that have this as a prefix will be
+                deleted.
+        Returns: `self`
+
+        Note:
+           It is OK if src_name and/or dest_name equals 'labels' or 'scores',
+           but these special attributes won't be deleted.
+        '''
+        assert src_name != dest_name
+        try:
+            value = getattr(self, src_name)
+            if src_name == 'labels':
+                value = value.clone()
+            setattr(self, dest_name, value)
+            try:
```

Comment:
If `src_name` is `scores`, you're deleting it, which does not match the `NOTE` given
in the comment.